src/components/common/TextField/TextFieldRefactor.tsx

```diff
@@ -0,0 +1,162 @@
+'use client';
+import React, {
+  ChangeEventHandler,
+  ForwardedRef,
+  useCallback,
+  useRef,
+  useState
+} from 'react';
+import * as styles from './TextFieldRefactor.css';
+import clsx from 'clsx';
+import { TextFieldRemoveIcon } from '@/asset/icons';
+import { Caption1, Caption2 } from '../Typography';
+import useForwardRef from '@/utils/useForwardRef';
+import { getStringOfValueLengthPerMax } from './utils/getStringOfValueLengthPerMax';
+
+/**
+ * props 타입, status 타입 정의
+ */
+interface TextFieldProps
+  extends Omit<React.HTMLProps<HTMLInputElement>, 'size'> {
+  error?: { message?: string };
+  size?: styles.InputSize;
+  helper?: string;
+  label?: string;
+  maxLength?: number;
+  defaultValue?: string | number;
+}
+
+/**
+ * TextField 컴포넌트
+ */
+const TextField = React.forwardRef<HTMLInputElement, TextFieldProps>(
+  function TextField(
+    {
+      name,
+      size = 'small',
+      helper,
+      error,
+      label,
+      onChange = () => {},
+      maxLength,
+      defaultValue,
+      ...inputProps
+    },
+    ref
+  ) {
+    if (!ref) throw Error(`${name}에 ref를 추가해주세요`);
+    const inputRef = useForwardRef<HTMLInputElement>(ref);
+    const lengthCountRef = useRef<HTMLDivElement>(null);
+    const [clearable, setClearable] = useState(false);
+
+    const clearInput = useCallback(() => {
+      if (!inputRef.current) return;
+      inputRef.current.value = '';
+      setClearable(false);
+
+      onChange({
+        target: inputRef.current
+      } as React.ChangeEvent<HTMLInputElement>);
+      inputRef.current.focus();
+    }, []);
+
+    const handleChange = useCallback<ChangeEventHandler<HTMLInputElement>>(
+      e => {
+        if (!(lengthCountRef.current && inputRef.current)) return;
+        onChange(e);
+        const value = e.target.value;
+        setClearable(value.length > 0);
+
+        lengthCountRef.current.innerText = getStringOfValueLengthPerMax(
+          inputRef.current.value,
+          maxLength
+        );
+      },
+      []
+    );
+
```

Comment:
상준님! 텍스트 필드 계속 수정하시느라 너무 고생 많으세요

브랜치 들어가서 확인해보았는데
다만 인풋 onBlur 시에만 에러 문구가 표출된다는 문제점을 발견했습니닷
콘솔 찍어서 확인해보니까 65행 얼리 리턴에서 막혀서 하단로직이 실행이 안되는 것 같네요?! 
하기처럼 onChange로 e.target.value 바인딩해주면 입력시에도 에러문구가 잘 뜨고 있는 것 확인햇는데 이부분 한번 체크해주시면 좋을 것 같습니닷

```suggestion
    const handleChange = useCallback<ChangeEventHandler<HTMLInputElement>>(
      e => {
      
      onChange?.({ target: { name, value: e.target.value } });
      
        if (!(lengthCountRef.current && inputRef.current)) return;
        const value = e.target.value;
        setClearable(value.length > 0);

        lengthCountRef.current.innerText = getStringOfValueLengthPerMax(
          inputRef.current.value,
          maxLength
        );
      },
      []
    );

```

Reply:
앗 제가 이부분을 놓쳤네요..!! 하영님이 말씀해주신 에러도 여기서 비롯되는것같습니다
maxLength prop이 없으면 ref도 없고 그러면 handleChange가 작동을 안하는 그런문제인거같네용
수정해서 다시 pr 올리겠습니다!!